spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsFunctionProcessor.java

```diff
@@ -151,11 +154,22 @@ public void orchestrateFunctionInvoking(ResolvableType resolvableType, String fu
 		Object[] adaptedInboundArguments = adaptAndRetrieveInboundArguments(stringResolvableTypeMap, functionName);
 		try {
 			if (resolvableType.getRawClass() != null && resolvableType.getRawClass().equals(Consumer.class)) {
+				//TOOD: Investigate why looking up by Consumer returns null
```

Comment:
Soby, there is nothing to investigate. It's a SCF issue that we simply need to document. Basically you can never have Consumer since after wrapping (which is what's in FunctionCatalog) Functions and Consumers are all Functions. So consumer is `Function<Flux<?>, Mono<Void>>`

Reply:
So you can technically remove this whole block of code